cli/blkio.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/alibaba/pouch/apis/types"
+	units "github.com/docker/go-units"
+)
+
+// WeightDevice defines weight device
+type WeightDevice struct {
+	values []*types.WeightDevice
+}
+
+func validateWeightDevice(val string) (*types.WeightDevice, error) {
+	pairs := strings.Split(val, ":")
+	if len(pairs) != 2 {
+		return nil, fmt.Errorf("bad format for weight device: %s", val)
+	}
+
+	weight, err := strconv.ParseUint(pairs[1], 10, 0)
+	if err != nil {
+		return nil, fmt.Errorf("invalid weight for device: %s", val)
+	}
+	if weight > 0 && (weight < 10 || weight > 1000) {
+		return nil, fmt.Errorf("invalid weight for device: %s", val)
+	}
+
+	return &types.WeightDevice{
+		Path:   pairs[0],
+		Weight: uint16(weight),
+	}, nil
+}
+
+// Set implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) Set(val string) error {
+	v, err := validateWeightDevice(val)
+	if err != nil {
+		return err
+	}
+	w.values = append(w.values, v)
+
+	return nil
+}
+
+// String implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) String() string {
+	var str []string
+	for _, v := range w.values {
+		str = append(str, fmt.Sprintf("%s:%d", v.Path, v.Weight))
+	}
+
+	return fmt.Sprintf("%v", str)
+}
+
+// Type implement WeightDevice as pflag.Value interface
+func (w *WeightDevice) Type() string {
+	return "value"
+}
+
+func (w *WeightDevice) value() []*types.WeightDevice {
+	var weightDevice []*types.WeightDevice
+	for _, v := range w.values {
+		weightDevice = append(weightDevice, v)
+	}
+
+	return weightDevice
+}
+
+// ThrottleBpsDevice defines throttle bps device
+type ThrottleBpsDevice struct {
+	values []*types.ThrottleDevice
+}
+
+func validateThrottleDeviceBps(val string) (*types.ThrottleDevice, error) {
+	pairs := strings.Split(val, ":")
+	if len(pairs) != 2 {
+		return nil, fmt.Errorf("bad format for throttle device: %s", val)
+	}
+
+	rate, err := units.RAMInBytes(pairs[1])
+	if err != nil || rate < 0 {
+		return nil, fmt.Errorf("invalid rate, rate must be positive. Corrent format should be <device-id>:<rate>, rate unit is optional and can be b/B, k/K, m/M... ")
```

Comment:
s/Corrent/Correct?

Can it be 0? `rate must be positive` is not equal to `!(rate<0)`.

Reply:
it can be 0 ,and I change the express